src/main/scala/io/iohk/ethereum/domain/Block.scala

```diff
@@ -0,0 +1,33 @@
+package io.iohk.ethereum.domain
+
+import io.iohk.ethereum.network.p2p.messages.CommonMessages
+import io.iohk.ethereum.network.p2p.messages.PV62.{BlockBody, BlockHeaderImplicits}
+import io.iohk.ethereum.rlp.{RLPEncodeable, RLPEncoder, RLPList, encode}
+
+/**
+  * This class represent a block as a header and a body which are returned in two different messages
+  *
+  * @param header Block header
+  * @param body   Block body
+  */
+case class Block(header: BlockHeader, body: BlockBody) {
+  override def toString: String = {
+    s"""BlockHeader {
+       | header: $header
+       | body: $body
+     """.stripMargin
+  }
+}
+
+object Block {
```

Comment:
Ideally Block should be under the 'blockchain' component (package). It's part of that interface along with BlockHeader and BlockBody. And the serialisation of the Block is also the responsibility of the Block component as it must know the internals of a Block (package blockchain.rlp?) with a dep on the ethereum.rlp package as a serialisation service.